File: ZG.Store.Domain/Models/OrderProduct.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace ZG.Store.Domain.Models
{
    public class OrderProduct
    {
        public int OrderProductId { get; set; }
        public int OrderId { get; set; }
        public Order Order { get; set; }

        public int ProductId { get; set; }
        public Product Product { get; set; }

        public int Quantity { get; set; }
        public decimal PricePerUnit { get; set; }
        public decimal TotalPrice { get; set; }
        public decimal? Discount { get; set; }
        public decimal Shipping { get; set; }
        [MaxLength(50)]
        public string DownloadKey { get; set; }
        [MaxLength(400)]
        public string DownloadURL { get; set; }
        public DateTime OrderDate { get; set; }
        public bool Active { get; set; }        
    }
}

File: ZG.Store.Admin/AngularApp/src/app/model/order-product.ts
export class OrderProduct{
    constructor(public orderProductId: number, public orderId: number, public productId: number,
        public productName: string, public quantity: number, public pricePerUnit: number, public totalPrice: number, 
        public shipping: number, public downloadKey: string, public downloadUrl: string, 
        public orderDate: string, public active: boolean, public discount?: number){}
}